File: dotnet/templates/be-CreateDto.cs
// Template: Create DTO
// Location: src/Modules/<ModuleName>/Dtos/Create<ModuleName>Dto.cs

using System.ComponentModel.DataAnnotations;
using Swashbuckle.AspNetCore.Annotations;

namespace YourApp.Modules.ModuleName.Dtos;

public class CreateModuleNameDto
{
    [Required]
    [SwaggerSchema("The name of the item")]
    public string Name { get; set; } = string.Empty;

    [SwaggerSchema("Optional description")]
    public string? Description { get; set; }

    // TODO: add more fields
    // Common annotations:
    // [StringLength(200)] public string Title { get; set; } = string.Empty;
    // [Range(1, 100)] public int Count { get; set; }
    // [EmailAddress] public string Email { get; set; } = string.Empty;
}

File: nestjs/templates/be-create.dto.ts
// Template: Create DTO
// Location: src/modules/<module-name>/dto/create-<module-name>.dto.ts

import { IsString, IsNotEmpty, IsOptional } from 'class-validator'
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger'

export class CreateModuleNameDto {
  @ApiProperty({ description: 'Name of the item', example: 'My item' })
  @IsString()
  @IsNotEmpty()
  name: string

  @ApiPropertyOptional({ description: 'Optional description', example: 'Some details' })
  @IsString()
  @IsOptional()
  description?: string

  // TODO: add more fields with class-validator + @ApiProperty decorators
  // Common decorators: @IsEmail(), @IsEnum(MyEnum), @IsNumber(), @IsBoolean(), @IsArray()
}
